CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -64,7 +64,7 @@ function Patient:onClick(ui, button)
     if self.message_callback then
       self:message_callback()
     else
-      local hospital = self.hospital or self.world:getLocalPlayerHospital()
+      local hospital = self.hospital or self.world:getHospital(self.tile_x, self.tile_y)
```

Comment:
But you presumably can't or shouldn't be able to vaccinate patients at other player's hospitals.